documentation/contributing/pullrequests.markdown

```diff
@@ -0,0 +1,85 @@
+
+ - Does the patch do what the author claims it does?
+ - Is it general enough to work with everyones use-case and not only the requestor's setup?
+ - Does it address all edge-cases you are aware of?
+ - If a new interface is introduced: Is this interface general enough for likely future modifications? Otherwise we might have to break API later on if we wish to add more useful features to the same module.
+ - Is the change [ABI compatible](https://en.wikipedia.org/wiki/Application_binary_interface) or is there a good reason (e.g. safety) to break it?
+ - Does the patch respect the [MoveIt! Code Style](code.html)?
+
+If (and only if) you answer all these questions with "yes", add a note saying "+1" to the request. This does **not** mean you like to have the claimed feature in MoveIt!. If it's not the first message you ever sent on a public channel concerning MoveIt, it means the next maintainer who reviews this request is free to merge it after review.
+If you are unsure about some part of the request, feel free to add a note and ask the author for details and explanations of individual changes.
+
+## Maintainer Side
+
+The ``*-devel`` branches of the repositories must always compile on the target platforms.
+
+The ``*-devel`` branches must not be force-pushed.
+
+### no direct commits
+
+No one is allowed to directly commit to the ``*-devel`` branches of the repositories. Instead they should create feature branches and add pull-requests.
+The only commits that might be pushed directly are cherry-picks from older branches (see below) and administrative changes (e.g. CHANGELOGS, tags).
+
+### 1+1 == merge
+
+If you see a trustworthy "+1" note in a request or the requestor is a fellow maintainer and all feedback has been addressed by the requestor,
+merge the request after your own review. Otherwise add a "+1" note after you are satisfied after the review.
+
+### squash-merge if possible
+
+Pull-requests with a single commit should *always* be squash-merged (github supports this via a drop-down list on the "merge" button).
+If a request contains "fixup commits" it should also be squash-merged. If you are unsure this is to the requestor's liking, ask him.
+
+### feature branches in upstream repositories
+
+In general, requests should originate from a private fork of the upstream repository. If maintainers expect support in the development of a patch, they are free to create a new branch in the upstream repository instead. This enables other maintainers to directly push changes there and enables users to add pull-requests targeting the feature branch. To keep the list of branches clear and unambiguous, **names for such branches should always follow the pattern ``request-<keyword description>``**. This makes it clear these branches are not relevant to users not involved in the respective request.
```

Comment:
I like to have descriptive keyword titles, but I also have branches display on every terminal prompt for git repos, so keeping it short is important to me. If we are just making up a new standard, I really like ``pr``. Could you also add the bit about distro name?